data-prepper-plugins/mutate-string-processors/README.md

```diff
@@ -0,0 +1,148 @@
+# Mutate String Processors
+The following is a list of processors to mutate a string.
+
+## SubstituteStringProcessor
+A processor that matches a field value against a regular expression and replaces all matches with a replacement string.
+
+### Basic Usage
+To start using Substitute String Processor with Data Prepper, create the following `pipeline.yaml`.
+
+```yaml
+pipeline:
+  source:
+    file:
+      path: "/full/path/to/logs_json.log"
+      record_type: "event"
+      format: "json"
+  processor:
+    - substitute_string:
+        entries:
+          - source: "source"
+            from: "from"
+            to: "to"
+          - source: "source2"
+            from: "from2"
+            to: "to2"
+  sink:
+    - stdout:
+```
+
+### Configuration
+* `entries` - (required) - A list of entries to add to an event
+    * `source` - (required) - The entry to be modified
```

Comment:
Please refer to elements of an event as a key. Entry is overloaded in the context of `entries` as well.

We should also inform our users what happens if "from" regex string does not match. Will this key still be modified or does it remain the same? Is there a log message or metric capturing the miss?